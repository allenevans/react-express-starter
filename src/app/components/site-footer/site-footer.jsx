import React from 'react';
import './site-footer.scss';

export const SiteFooter = () => (
  <footer className="site-footer">
    Footer
  </footer>
);

export default SiteFooter;
