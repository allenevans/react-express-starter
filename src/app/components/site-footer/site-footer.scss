@import '../../../client/breakpoints';
@import '../../../client/variables';

.site-footer {
  flex: 0 0 auto;
  text-align: center;

  padding: .5rem $GUTTER_WIDTH_MOBILE;
  @include for-tablet-portrait-up {
    padding: 1rem $GUTTER_WIDTH_TABLET;
  }
}
