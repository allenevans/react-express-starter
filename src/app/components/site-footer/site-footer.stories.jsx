import React from 'react';
import { storiesOf } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';
import { withViewport } from '@storybook/addon-viewport/preview';
import { SiteFooter } from './site-footer';

const siteFooterStories = storiesOf('Site Footer', module);

const styleBlock = <style type="text/css">{`
  body {
    background-color: #ccc;
    height: 100vh;
    display: flex;
    flex-direction: column;
  }

  #container {
    display: flex;
    flex-direction: column;
    height: 100vh;
  }

  .content {
    flex: 1 0 auto;
  }
`}
</style>;

siteFooterStories
  .addDecorator(withViewport())
  .addDecorator(story => <MemoryRouter>{story()}</MemoryRouter>)
  .add('Default footer', () => <React.Fragment>
    {styleBlock}

    <div id="container">
      <div className="content">Footer at the bottom of the page</div>
      <SiteFooter/>
    </div>
  </React.Fragment>);
