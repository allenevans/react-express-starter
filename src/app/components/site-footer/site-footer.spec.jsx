import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { SiteFooter } from './site-footer';

const renderer = new ReactShallowRenderer();

describe('site-footer component', () => {
  it('should render the component', () => {
    const tree = renderer.render(<SiteFooter />);

    expect(tree).toMatchSnapshot();
  });
});
