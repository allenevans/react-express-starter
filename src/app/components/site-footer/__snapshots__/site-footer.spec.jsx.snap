// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`site-footer component should render the component 1`] = `
<footer
  className="site-footer"
>
  Footer
</footer>
`;
